nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -124,28 +135,22 @@ class BtActionNode : public BT::CoroActionNode
       throw std::runtime_error("Goal was rejected by the action server");
     }
 
-    auto future_result = action_client_->async_get_result(goal_handle_);
-    rclcpp::executor::FutureReturnCode rc;
-    do {
-      rc = rclcpp::spin_until_future_complete(node_, future_result, server_timeout_);
-      if (rc == rclcpp::executor::FutureReturnCode::TIMEOUT) {
```

Comment:
does `server_timeout_` still serve a purpose? It seems like `on_wait_for_result` doesn't have a well-defined period of execution, whereas `on_server_timeout` did.

Reply:
Server timeout does not serve a purpose anymore. That was because we were spinning until successor timeout. Now we just check on a boolean and then return -- which is much faster (100ms -> less than 10ms). The Bt itself is throttled to 10hz so there's no issues there.

Reply:
Okay great there is throttling. I think it's best to remove the `server_timeout_` blackboard initialization up above just for clarity's sake. The change looks good to me.

Reply:
Good suggestion -- I'll do that